controllers/mirrorpeer_controller.go

```diff
@@ -749,6 +733,13 @@ func (r *MirrorPeerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&multiclusterv1alpha1.MirrorPeer{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&addonapiv1alpha1.ManagedClusterAddOn{}, builder.WithPredicates(predicate.NewPredicateFuncs(func(object client.Object) bool {
+			mca, ok := object.(*addonapiv1alpha1.ManagedClusterAddOn)
+			if !ok || mca.Name != setup.TokenExchangeName {
+				return false
+			}
```

Comment:
This check `mca.Name != setup.TokenExchangeName` shouldn't be required since we are setting the controller reference ourselves.  We would want the reconcile to be triggered even if the name does not match right.

Reply:
There are chances of it being changed erroneously by some user or automation. This just ensures we're reconciling the right thing.